### PR TITLE
Basis transfer code

### PR DESCRIPTION
**Pull request by floatingCatty**
1. Add a new loss function: EigHamLoss

2. Add a new eigenvalue and eigenvector calculation module: I want to use the eigenvector constraint, instead of the Hamiltonian and overlap to bind the model from overfitting, but it fails. The module still works.

### CODE REVIEW

File: dptb/data/AtomicData.py [hunk 1]
```diff
@@ -104,6 +104,7 @@
     AtomicDataDict.HAMILTONIAN_KEY, # new # should be nested
     AtomicDataDict.OVERLAP_KEY, # new # should be nested
     AtomicDataDict.ENERGY_EIGENVALUE_KEY, # new # should be nested
+    AtomicDataDict.EIGENVECTOR_KEY, # new # should be nested
```
Comment (QG-phy) [inline dptb/data/AtomicData.py:107]: 是想把eigenvector 加入训练标签吗？估计不行啊。求导会出问题。

Comment (floatingCatty) [reply dptb/data/AtomicData.py:107]: 这块是deeptb接dftio算实空间LDOS的时候用的

Comment (QG-phy) [reply dptb/data/AtomicData.py:107]: ok

**Review by QG-phy — APPROVED** [review]
